core/txpool/legacypool/legacypool.go

```diff
@@ -605,33 +609,40 @@ func (pool *LegacyPool) validateTx(tx *types.Transaction) error {
 	return pool.validateAuth(tx)
 }
 
+// limitTxFromDelegated ensures the account with either delegation or pending
+// delegation can at most send one inflight **executable** transaction.
+func (pool *LegacyPool) limitTxFromDelegated(tx *types.Transaction) error {
+	from, _ := types.Sender(pool.signer, tx) // validated
+
+	// Short circuit if the sender has neither delegation nor pending delegation.
+	if pool.currentState.GetCodeHash(from) == types.EmptyCodeHash && len(pool.all.auths[from]) == 0 {
+		return nil
+	}
+	pending := pool.pending[from]
+	if pending == nil {
+		// Transaction with gapped nonce is not supported for delegated accounts
+		if pool.pendingNonces.get(from) != tx.Nonce() {
+			return ErrOutOfOrderTxFromDelegated
+		}
+		return nil
+	}
+	// Transaction replacement is supported
+	if pending.Contains(tx.Nonce()) {
+		return nil
+	}
+	if pending.Len() >= 1 {
```

Comment:
Isn't this always the case? if no pending transaction is in pool.pending[from], we will return in line 627.
In this case the length of pending will always be >= 0, no?

Reply:
yeah, should be..